### PR TITLE
Feature/16 add Application

### LINKED ISSUE #1: Implement integration of AqualityServices with Dependency Injection container
- [x] Add a simple service, e.g. Logger or ElementActionRetrier
- [x] Implement integration with DI container
- [x] Implement AqualityServices class with static access to registered services

### LINKED ISSUE #16: Add IApplication interface and add AppContainer to AqualityServices

### PR DESCRIPTION
**Pull request by mialeska**
### PR Details

Add IApplication to AqualityServices
Add guava exclusion for guice to fix the conflict with selenium library
Added tests for Application
Refactor AqualityServices
Update azure-pipelines.yml

##### Related Issue Link: 

closes #16, closes #1 

##### How Has This Been Tested
Added tests to start browser and simple desktop application

##### Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/test/java/tests/application/browser/BrowserTests.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    @Test
+    public void testShouldBePossibleToGetDriver() {
+        Assert.assertNotNull(AqualityServices.getApplication().getDriver(),
+                "should be possible get driver from the application");
+    }
+
+    @Test
+    public void testShouldBePossibleToSetImplicitWait() {
+        AqualityServices.getApplication().setImplicitWaitTimeout(1, TimeUnit.SECONDS);
```
Comment (knysh) [inline src/test/java/tests/application/browser/BrowserTests.java]: Would it better to wrap this into try\catch and use asser.fail in catch block?
I think test should have assert and some message in fail case.

File: src/test/java/tests/application/windowsApp/ApplicationTests.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    @Test
+    public void testShouldBePossibleToGetDriver() {
+        Assert.assertNotNull(AqualityServices.getApplication().getDriver(),
+                "should be possible get driver from the application");
+    }
+
+    @Test
+    public void testShouldBePossibleToSetImplicitWait() {
+        AqualityServices.getApplication().setImplicitWaitTimeout(1, TimeUnit.SECONDS);
```
Comment (knysh) [inline src/test/java/tests/application/windowsApp/ApplicationTests.java]: Would it better to wrap this into try\catch and use asser.fail in catch block?
I think test should have assert and some message in fail case.

Comment (mialeska) [reply src/test/java/tests/application/windowsApp/ApplicationTests.java]: fixed

File: src/test/java/tests/application/browser/BrowserTests.java (hunk 1 above)
Comment (mialeska) [reply src/test/java/tests/application/browser/BrowserTests.java]: fixed

File: src/main/java/aquality/selenium/core/application/AqualityServices.java [hunk 3]
```diff
@@ -2,43 +2,61 @@
 
 import com.google.inject.Guice;
 import com.google.inject.Injector;
+import com.google.inject.Provider;
+
+import java.util.function.Function;
+import java.util.function.Supplier;
 
 /**
  * Describes static methods which help work with Application and dependency injector.
```
[the remark this answers is not in the document]
Comment (mialeska) [reply src/main/java/aquality/selenium/core/application/AqualityServices.java]: thanks, fixed

**Pull request merged**